log/context.go

```diff
@@ -56,7 +56,13 @@ func New(sink io.Writer, typ Type, level Level) *Logger {
 		logger = zerolog.New(zerolog.ConsoleWriter{Out: sink})
 	}
 
-	logger = logger.Level(level).With().Timestamp().Logger()
+	logger = logger.
+		Level(level).
+		With().
+		Timestamp().
+		Logger().
+		Hook(zerolog.HookFunc(scopeHook))
```

Comment:
this incurs a check for the scope context key on each logger, regardless of whether it uses scope or not. another design would be to inject the hook upon calling `Scoped`. that might be an unnecessary optimization though hehe